src/services/cardService.ts

```diff
@@ -3,78 +3,33 @@ import Card, { CardDocument } from '../models/card';
 import util from '../modules/util';
 import { Types } from 'mongoose';
 import { CardResponseDto } from '../intefaces/CardResponseDto';
+import BestCard from '../models/bestCard';
 interface CardIdAndCnt {
   _id: Types.ObjectId;
   count: number;
 }
 
-const findBestDummy = async (
-  userId: Types.ObjectId | undefined,
-  size: number
-) => {
-  const topics = [
-    'SOPT에서 얻은\n가장 소중한 인연은?',
-    '절친 동생이랑 연애\nVS\n동생 절친이랑 연애',
-    '내가 선호하는 주종과 \n나의 주량은?',
-    '사생활 노출\nVS\n노출로 생활',
-    '만약 힘들게 취업한 회사의\n상사가 전남친이라면?',
-    '만약 내가 다른\nMBTI가 될 수 있다면?',
-    '상대방의 mbti를\n맞춰보세요!',
-    '고양이상과 강아지상 중, \n나의 취향은?',
-    '사주나 신점을 \n믿는다 VS 안믿는다',
-    '여기서 결혼을 제일\n빨리 할 것 같은 사람은?',
-    '이 중 썸탄 경험이\n많은 사람은\n누구인가요?',
-    '매일 아침, 눈을 뜬 다음\n나의 모닝 루틴은?',
-    '나만 아는 숨겨진 맛집!\n한 곳만 알려주실래요?',
-    '서투른 애인\nVS\n능숙한 애인',
-    '연애할 때, 중요도는\n성격 VS 외모',
-    '사랑과 우정 중\n하나를 택해야한다면?\n당신의 선택과 이유는?',
-    '주변환경과 본인의 의지\n둘 중 무엇이 \n삶에 더 큰 영향을\n끼친다고 생각하세요?',
-    '최고의 복수는\n성공이라는 말에\n동의하시나요?\n그 이유는?',
-    '시간을 돌릴 수 있다면, \n언제로 돌아가고 싶은가요?',
-    '발표를 망쳤을 때\n위로와 격려받기\nVS \n냉철한 피드백 받기',
-    '내가 생각하는 이상적인 애인 \n또는 연애에 대해 말해주세요',
-    '여기서 차기 파트장\n할 것 같은 사람 고르기',
-    '자신의\n인생 영화작\n말해주기',
-    '결혼을 하고 싶나요?\n만약 한다면, 언제쯤?',
-    '상대방을 보면 떠오르는 \n색깔을 말해주세요!',
-    '최애가수 1명과\n그 이유를\n말해주세요!',
-    '싸웠을 때,\n시간이 필요한가요\n당장 풀기를 원하나요?',
-    '이성친구 립밤 빌려주는 애인\nVS\n이성친구 빨대 같이쓰는 애인'
-  ];
-  let count = 0;
-  const dummyList: CardResponseDto[] = [];
-  for (const item of topics) {
-    const dummyCard = await Card.find({
-      content: item
-    });
-    console.log(item);
-    dummyList[count] = {
-      _id: dummyCard[0]._id,
-      content: dummyCard[0].content,
-      tags: dummyCard[0].tags,
-      category: dummyCard[0].category,
-      filter: dummyCard[0].filter
-    };
-    count++;
-  }
-  const data = await Promise.all(
-    dummyList.map(async (item: any) => {
-      const isBookmark =
-        (await Bookmark.find({ user: userId, card: item._id }).count()) > 0
-          ? true
-          : false;
+const findBestDummy = async (userId: Types.ObjectId | undefined) => {
+  const bestCards = await BestCard.find();
+
+  const cardResList: (CardResponseDto | null)[] = await Promise.all(
```

Comment:
Nullable 타입은 다양한 곳에서 쓰일 것 같아서! 공용으로 쓰는 타입을 지정해두는 곳이 있다면 null이될 수 있는 값에 대해서는 이런 식으로 지정해놔도 괜찮을 것 같아요 !
- types.d.ts
```typescript 
export type Nullable<T> = T | null;
```
```suggestion
  const cardResList: Nullable<CardResponseDto> = await Promise.all(
```

Reply:
@Nahee-Park 오오 맞아요 안그래도 엄청 자주 쓰고 있었거든요
저렇게 하면 가독성도 좋고 편리할 것 같아요!
그렇담 저 함수 파라미터 userId 처럼 undefined인 경우도 허용하고 싶을 때도 똑같이 적용 할 수 있을까요?

Reply:
@kyY00n  넹 undefined일 때에도 type Undefinedable 만들어서 사용할 수 있을 것 같아여 !

Reply:
맞아 그리구 d.ts 파일 확장자 쓰면 공용으로 쓰는 타입들 그때마다 임포트하지 않아도 전역적으로 커스텀 타입들 쓸 수 있는데 컴파일 옵션이라던가 추가적으로 좀 설정해줘야 할 것들이 있어서 관심 있으면 찾아봐도 좋을 것 같아여 !

Reply:
와아아 이걸 지금봤네ㅠ 적용해봐야겠으요 ㅎㅎㅎ 감사합니당